From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tariq Bashir <120014322+t-bashir-bs@users.noreply.github.com>
Date: Wed, 4 Dec 2024 15:43:31 +0000
Subject: OS-15658: do not convert tooltip position to screen coordinates for
 wayland

Weston wayland draws popup windows and tooltips in the parent window's coordinate space
and not in screen space. This patch ensures that the tooltip position is not converted to
screen coordinates when running on wayland so its position is correct.

diff --git a/ui/views/corewm/tooltip_aura.cc b/ui/views/corewm/tooltip_aura.cc
index 40dc7f595665d74956bded61e52c8f0d1ce20bc2..f447e3dc03c37a51a77f6c304c556dd96f86527c 100644
--- a/ui/views/corewm/tooltip_aura.cc
+++ b/ui/views/corewm/tooltip_aura.cc
@@ -248,10 +248,13 @@ void TooltipAura::Update(aura::Window* window,
 
   // Convert `position` to screen coordinates.
   gfx::Point anchor_point = position;
+  // Weston wayland compositor doesn't want tooltip co-ordinates in screen space.
+#if !BUILDFLAG(IS_OZONE_WAYLAND)
   aura::client::ScreenPositionClient* screen_position_client =
       aura::client::GetScreenPositionClient(window->GetRootWindow());
   CHECK(screen_position_client);
   screen_position_client->ConvertPointToScreen(window, &anchor_point);
+#endif
 
   new_tooltip_view->SetMaxWidth(GetMaxWidth(anchor_point));
   new_tooltip_view->SetText(tooltip_text);
